src/ucp/core/ucp_request.inl

```diff
@@ -395,13 +395,15 @@ ucp_request_send_state_init(ucp_request_t *req, ucp_datatype_t datatype,
 
     switch (datatype & UCP_DATATYPE_CLASS_MASK) {
     case UCP_DATATYPE_CONTIG:
+        req->send.state.dt.dt.contig.memh       = NULL;
```

Comment:
do we still need to keep md_map field in dt.contig?
what is the relation between dt.contig.md_map and dt.contig.memh->md_map?

Reply:
ucp_rma_basic_progress_get/put still using it, not sure if it's right

Reply:
ahh - ucp_rndv_rts_pack()

Reply:
maybe move it under "rndv" field in ucp_request_t::send?
I see this md_map is being used/initialized in non-rndv places as well, probably it's not needed

Reply:
md_map is not used out of rndv flow 
but ucp_request_t::send::state initialization is common code 
same is about ucp_request_t::recv::state - for ucp_rndv_rtr_pack() we may add md_map under ucp_request_t::recv::am, but ::recv::state is being initialized from ucp_rndv_receive() which is common for tag and am

Reply:
what if we add md_map to https://github.com/openucx/ucx/blob/a1492f96ba2721b766e8a4739294a263beb50244/src/ucp/core/ucp_request.h#L242 for rndv RTS, and https://github.com/openucx/ucx/blob/a1492f96ba2721b766e8a4739294a263beb50244/src/ucp/core/ucp_request.h#L298 in case of RTR (so it will be in "rndv_req" and not "rreq")
I assume we need md map for rndv to know which remote keys to pack?